src/mpid/ch4/netmod/ucx/ucx_am.h

```diff
@@ -11,52 +11,85 @@
 
 #include "ucx_impl.h"
 
+#define MPIDI_UCX_SHORT_DATA_SZ  16 * 1024
+
 MPL_STATIC_INLINE_PREFIX void MPIDI_UCX_am_isend_callback(void *request, ucs_status_t status)
 {
     MPIDI_UCX_ucp_request_t *ucp_request = (MPIDI_UCX_ucp_request_t *) request;
-    MPIR_Request *req = ucp_request->req;
-    int handler_id = req->dev.ch4.am.netmod_am.ucx.handler_id;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_UCX_AM_ISEND_CALLBACK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_UCX_AM_ISEND_CALLBACK);
 
-    MPL_free(req->dev.ch4.am.netmod_am.ucx.pack_buffer);
-    req->dev.ch4.am.netmod_am.ucx.pack_buffer = NULL;
-    MPIDIG_global.origin_cbs[handler_id] (req);
-    ucp_request->req = NULL;
+    if (ucp_request->u.am_send.pack_buffer) {
+        MPL_free(ucp_request->u.am_send.pack_buffer);
+    }
+    if (ucp_request->u.am_send.sreq) {
+        MPIDIG_global.origin_cbs[ucp_request->u.am_send.handler_id] (ucp_request->u.am_send.sreq);
+    }
+    MPIDI_UCX_UCP_REQUEST_RELEASE(ucp_request);
 
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_UCX_AM_ISEND_CALLBACK);
 }
 
-MPL_STATIC_INLINE_PREFIX void MPIDI_UCX_am_send_callback(void *request, ucs_status_t status)
+/* internal routine for sending data. `send_buf` will be freed on completion. If there is
+   `payload` (when not NULL), an iov of length 2 will be used to avoid extra copy. */
+MPL_STATIC_INLINE_PREFIX int MPIDI_ucx_am_send_internal(int rank, MPIR_Comm * comm, int handler_id,
```

Comment:
Same. I want to distinguish from other `MPIDI_UCX_` functions to signify that this function is an internal function.

Reply:
Is this something we do for other similar cases?

Reply:
No, I think this is new practice from me :). Knowing a function or a structure is static or internal to a file is very helpful in reading a code -- so that one is assured that he doesn't need to check other files for completeness and changes can be made consistently within a file. Unfortunately since these are in a header file, we still need prefix to avoid name collisions. 

I guess I could stick to `MPIDI_UCX_` in this PR, then follow up with a separate PR for prefix change since it is starting a new convention and probably deserve more agreement :)

Reply:
I agree with you that differentiating local functions from the others improves readability. However, you are adding the `internal` suffix to one of the functions' name and also changing the namespace. I think adding the `internal` (or `local`) suffix is fine but for now sticking to current namespace would be preferable instead of adopting new conventions unilaterally : )

Reply:
Will fix.

Reply:
updated